node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -421,6 +440,11 @@ impl<C: Context> ActiveValidator<C> {
             signature,
         )]))
     }
+
+    /// Returns a panorama that is valid to use in our own unit at the given timestamp.
+    fn panorama_at(&self, state: &State<C>, timestamp: Timestamp) -> Panorama<C> {
+        state.to_valid_panorama(self.vidx, state.panorama().cutoff(state, timestamp))
```

Comment:
Can you extract `state.panorama().cutoff(state, timestamp)` to a value and comment why we're doing the cutoff? I must admit that over the last months I've had to read the code at least three times to remind myself why we need to do it.